File: rust/src/models/metric_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{event_id::EventId, event_type::EventType, metric_event_data::MetricEventData};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MetricEvent {
    #[serde(flatten)]
    pub flatten_metriceventdata: MetricEventData,

    pub id: EventId,

    pub timestamp: String,

    pub r#type: EventType,
}

impl MetricEvent {
    pub fn new(id: EventId, timestamp: String, r#type: EventType) -> Self {
        Self {
            flatten_metriceventdata: Default::default(),
            id,
            timestamp,
            r#type,
        }
    }
}

File: typescript/src/models/metricEvent.ts
// this file is @generated
import { type EventId, EventIdSerializer } from "./eventId";
import { type EventType, EventTypeSerializer } from "./eventType";
import { type MetricEventData, MetricEventDataSerializer } from "./metricEventData";

export interface MetricEvent {
  flattenMetriceventdata: MetricEventData;

  id: EventId;

  timestamp: Date;

  type: EventType;
}

export const MetricEventSerializer = {
  _fromJsonObject(object: any): MetricEvent {
    return {
      flattenMetriceventdata: MetricEventDataSerializer._fromJsonObject(
        object["__flatten_metriceventdata"]
      ),
      id: EventIdSerializer._fromJsonObject(object["id"]),
      timestamp: new Date(object["timestamp"]),
      type: EventTypeSerializer._fromJsonObject(object["type"]),
    };
  },

  _toJsonObject(self: MetricEvent): any {
    return {
      __flatten_metriceventdata: MetricEventDataSerializer._toJsonObject(
        self.flattenMetriceventdata
      ),
      id: EventIdSerializer._toJsonObject(self.id),
      timestamp: self.timestamp,
      type: EventTypeSerializer._toJsonObject(self.type),
    };
  },
};
